src/gui/generalsettings.cpp

```diff
@@ -327,26 +329,15 @@ void GeneralSettings::slotUpdateChannelChanged(const QString &translatedChannel)
     };
 
     const auto updateChannelFromLocalized = [](const int index) {
-        auto channel = QString{};
-
-        switch (index)
-        {
-        case 0:
-            channel = QStringLiteral("stable");
-            break;
-        case 1:
-            channel = QStringLiteral("beta");
-            break;
-        default:
-            channel = QString{};
+        if (index == 1) {
+            return QStringLiteral("beta");
```

Comment:
There is no "daily" update channel selectable by users or supported by the client (unless you mean if we would like to add one in the code in the future)?

Reply:
my point is about future proofing the code
while fixing an issue, we may also improve the code (if that is quick to do and only if that is quick todo)

Reply:
I don't see a quick way to improve this, a proper fix would give us a proper identifier for the user selection rather than trying to decode it via the index number

Until we create a custom widget (or eventually QML component) that does this, I would say let's do it like this with safe defaults and restrictive handling of options as this bug is hitting real users

Reply:
@claucambra Have you considered putting the channel name and its localized string into an array of pairs (`"channel name" : "localized string"`) and then just filling the `updateChannel` with what's in that array? then, adding a `daily` would just imply adding an entry to that array, or is it something that will take a lot of time? if this is the case, we can skip it